### PR TITLE
NO-ISSUE: add build tags option

### PR DESCRIPTION
**Pull request by rccrdpccl**
This would enable us to pass BUILD_TAGS from downstream build to produce, for instance, FIPS-enable artifacts

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by gamli75 — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -218,10 +220,10 @@ build-in-docker:
 
 build-assisted-service:
 	# We need the CGO_ENABLED for the go-sqlite library build.
-	cd ./cmd && CGO_ENABLED=1 go build $(DEBUG_ARGS) -o $(BUILD_FOLDER)/assisted-service
+	cd ./cmd && CGO_ENABLED=1 go build -o $(GO_BUILD_TAGS) $(DEBUG_ARGS) -o $(BUILD_FOLDER)/assisted-service
```
Comment (omertuc) [inline Makefile:223]: having two `-o` flags means the first one is simply ignored.

**Review by adriengentil — APPROVED** [review]

**Pull request merged**